// @lib/auth.js
import cookie from 'js-cookie'

export const handleSession = (res) => {
  if (res.status === 440) {
    // flash('Session has expired.', 'danger')
    userLogout()
    setTimeout(() => {
      window.location.replace('/')
    }, 1000)
  }
}

export const userLogout = () => {
  removeCookie('token')
  removeLocalStorage('user')
}

export const setCookie = (key, value) => {
  cookie.set(key, value, {expires: 1, sameSite: 'strict'})
}

export const removeCookie = (key) => {
  cookie.remove(key)
}

export const removeLocalStorage = (key) => {
  localStorage.removeItem(key)
}

export const getCookie = (key) => {
  return cookie.get(key)
}

export const setLocalStorage = (key, value) => {
  localStorage.setItem(key, JSON.stringify(value))
}

export const authenticate = (data, next) => {
  setCookie('token', data.token)
  setTimeout(async () => {
    delete data['token']
    setLocalStorage('user', data)
    return next()
  }, 200)
}

export const isAuth = () => {
  if (getCookie('token')) {
    if (localStorage.getItem('user')) {
      return JSON.parse(localStorage.getItem('user'))
    }
  }
  return false
}
