// api/api.js
import axios from 'axios'

async function send ({ method, path, data, token }) {
  try {
    const response = await axios({
      method,
      baseURL: process.env.API_BASE_URL,
      url: path,
      data,
      headers: {
        ...(token ? { Authorization: `Bearer ${token}` } : {})
      }
    })
    return new Promise(resolve => resolve(response.data))
  } catch (error) {
    if(error.response){
      throw error.response.data
    }
    if(error.message === "Network Error"){
      new Error('Network Connection Error, Please try later.')
    }
    throw error.message
  }
}

export function post (path, data, token = null) {
  return send({ method: 'post', path, data, token })
}

export function put (path, data, token = null) {
  return send({ method: 'put', path, data, token })
}

export function patch (path, data, token = null) {
  return send({ method: 'patch', path, data, token })
}

export function del (path, token = null) {
  return send({ method: 'delete', path, token })
}

export function get (path, data = {}, token = null) {
  const params = Object.keys(data).map(key => `${key}=${data[key]}`).join('&')
  return send({
    method: 'get',
    path: `${path}${params ? '?' + params : ''}`,
    token
  })
}
