// api/user.js
import * as api from './api'

export function register (data) {
  return api.post('/user/register', data)
}

export function login (data) {
  return api.post('/user/login', data)
}

export function logout (data) {
  return api.get('/user/logout', data)
}

export function getProfile (data, token) {
  return api.post('/user/account', data, token)
}

export function getUsers (page, data, token) {
  return api.get(`/admin/users/${page}`, data, token)
}

export function getUser (id, data, token) {
  return api.get(`/admin/user/${id}`, data, token)
}

export function getUserCount (data, token){
  return api.get('/admin/user-count', data, token)
}

export function update (data, token) {
  return api.patch('/user/account', data, token)
}

export function updateSettings (id, data, token) {
  return api.patch(`/admin/settings/${id}`, data, token)
}

export function deleteUser (data, token) {
  return api.post('/user/delete', data, token)
}

export function forgotPassword (data, token) {
  return api.post('/user/forgot', data, token)
}

export function userResetPassword (data, token) {
  return api.post('/user/reset-password', data, token)
}

export function passwordUpdate (data, token) {
  return api.post('/user/password-update', data, token)
}

export function submitQuote(data, token){
  return api.post('/quote', data, token)
}
